### PR TITLE
adding includingWith: , matching:  and startingWith: methodes. fixes #5437

### LINKED ISSUE #5437: Add Package queries
We should add two query methods to `RPackageOrganizer`:

```smalltalk
organiser selectPackages: [ :each | ... ]
```

```smalltalk
organiser selectPackageWithName: [ :name | ...  ]
```

On top of those, users could easily build more complex queries (beginsWith, endsWith, substrings, regex-based...)

### PR DESCRIPTION
**Pull request by MoussaSak**
adding includingWith: , matching: and startingWith: methodes. fixes #5437

### CODE REVIEW

**Review by tesonep — CHANGES_REQUESTED** [review]
@MoussaSak It is possible to add tests to it.
Thanks

**Comment by guillep** [comment]
Hi @MoussaSak, what is the status of this PR?

**Comment by MoussaSak** [comment]
Hello @tesonep and @guillep, okey i will add some tests on it.

File: src/RPackage-Core/RPackageOrganizer.class.st [hunk 1]
```diff
@@ -451,7 +451,7 @@ RPackageOrganizer >> hasRegistered [
 	^ (actionSequence isNil not) and: [(MCWorkingCopy myDependents includes: self)]"
 ]
 
-{ #category : #testing }
+{ #category : #accessing }
```
Comment (estebanlm) [inline src/RPackage-Core/RPackageOrganizer.class.st:454]: no this is bad. 
correct protocol is "testing" not "accessing"

File: src/RPackage-Core/RPackageOrganizer.class.st [hunk 2]
```diff
@@ -463,12 +463,19 @@ RPackageOrganizer >> includesPackageBackPointerForClass: aClass [
 	^ classPackageMapping includesKey: aClass instanceSide name asSymbol
 ]
 
-{ #category : #testing }
+{ #category : #accessing }
```
Comment (estebanlm) [inline src/RPackage-Core/RPackageOrganizer.class.st:466]: same this one.

File: src/RPackage-Core/RPackageOrganizer.class.st [hunk 3]
```diff
@@ -463,12 +463,19 @@ RPackageOrganizer >> includesPackageBackPointerForClass: aClass [
...
 RPackageOrganizer >> includesPackageNamed: aSymbol [
 	"Answer whether the receiver get a package named aSymbol as registered."
 	^ packages includesKey:  aSymbol asSymbol
 ]
 
+{ #category : #'private testing' }
+RPackageOrganizer >> includingWith: aString [
```
Comment (estebanlm) [inline src/RPackage-Core/RPackageOrganizer.class.st:473]: "includingWith" has no meaning in English :)
`includesPackageWith:` ?

File: src/RPackage-Core/RPackageOrganizer.class.st [hunk 4]
```diff
@@ -888,6 +903,13 @@ RPackageOrganizer >> signalPackageExists: aPackageName [
 	RPackageConflictError signal: ('A package named {1} already exists' format: {aPackageName})
 ]
 
+{ #category : #accessing }
+RPackageOrganizer >> startingWith: aString [
```
Comment (estebanlm) [inline src/RPackage-Core/RPackageOrganizer.class.st:907]: in other thread @guillep suggested to use `packagesStartingWith:` instead.

**Review by estebanlm — CHANGES_REQUESTED** [review]

**Comment by guillep** [comment]
I'll close this, as this is duplicate from #5905

**Pull request closed without merging**